redis/client.py

```diff
@@ -1443,6 +1477,11 @@ def psubscribe(self, *args, **kwargs):
         # for the reconnection.
         new_patterns = self._normalize_keys(new_patterns)
         self.patterns.update(new_patterns)
+        if not self.subscribed:
+            # Set the subscribed_event flag to True
+            self.subscribed_event.set()
+            # Clear the health check counter
+            self.health_check_response_counter = 0
```

Comment:
Should this be done at the end of `clean_health_check_response`?

Reply:
I don't want to put it there because the following scenario is possible:
1. p.subscribe("foo")
2. health check is performed
3. p.unsubscribe("foo")
4. a health check response still hasn't received
5. p.unsuscribe("foo") 
6. clean_health_check_response is being called by the unsubscribe command, the health check  response hasn't arrived yet and it exists the loop due to ttl runs-out
7. the health check response only now received 
8. p.subscribe() is being called - self.subscribed is still False so a health check will be performed and we should clean the existing health check response before we continue. 
9. If we add 'clean_health_check_response=0' at the end of clean_health_check, we will clean the counter in step 6, so we won't be able to clean the socket from the response on step 8.